test/diff-test.html

```diff
@@ -0,0 +1,128 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Enhance Custom Element Mixin DOM Diffing Test</title>
+  <link rel="icon" href="data:image/svg+xml,<svg xmlns=%22http://www.w3.org/2000/svg%22 viewBox=%220 0 100 100%22><text y=%22.9em%22 font-size=%2290%22>✨</text></svg>">
+</head>
+<body>
+
+  <my-heading>
+    <button>Submit</button>
+  </my-heading>
+
+  <script type="module">
+    import { runTests } from '@web/test-runner-mocha'
+    import { assert } from '@esm-bundle/chai'
+    import BaseElement from '@enhance/base-element'
+    import TemplateMixin from '@enhance/template-mixin'
+    import CustomElementMixin from '../index.mjs'
+    import morphdom from 'morphdom'
+
+    class CustomElement extends CustomElementMixin(TemplateMixin(BaseElement)) {}
+
+    const MorphdomMixin = (superclass) => class extends superclass {
+      constructor(args) {
+        super(args)
+        this.process = this.process.bind(this)
+      }
+
+      attributeChangedCallback(name, oldValue, newValue) {
+        if (oldValue !== newValue) {
+          this.process()
+        }
+      }
+
+      process() {
+        const tmp = this.render({
+          html: this.html,
+          state: this.state
+        }).trim()
+
+        const updated = document.createElement('div')
+        updated.innerHTML = tmp
+        this.scrubTemplate(updated)
+
+        const root = this.shadowRoot
+          ? this.shadowRoot
+          : this
+
+        const slotEl = document.createElement('div')
```

Comment:
These additions will be added to the morph mixins as well. 
Once they land in the morphdommixin I will use that in these tests.